### PR TITLE
Small improvements in conversion generator

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: pkg/runtime/conversion.go [hunk 1]
```diff
@@ -40,13 +40,13 @@ func JSONKeyMapper(key string, sourceTag, destTag reflect.StructTag) (string, st
...
-	convertStringSliceToBool,
-	convertStringSliceToInt64,
+	Convert_Slice_string_To_string,
+	Convert_Slice_string_To_int,
+	Convert_Slice_string_To_bool,
+	Convert_Slice_string_To_int64,
 }
 
-func convertStringSliceToString(input *[]string, out *string, s conversion.Scope) error {
+func Convert_Slice_string_To_string(input *[]string, out *string, s conversion.Scope) error {
```
Comment (lavalamp) [inline pkg/runtime/conversion.go]: This throws away every index except 0? Is that useful?

**Comment by lavalamp** [comment]
Maybe the slice to single value converters should error on lists > 1 element?

File: pkg/runtime/conversion.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/runtime/conversion.go]: I added returning error for lists with 1+ elems.

**Comment by wojtek-t** [comment]
Agree & done (although slightly unrelated to this PR).

@lavalamp - PTAL

**Comment by wojtek-t** [comment]
@lavalamp - it seems that we are leveraging this behavior:
"Error from server: converting slice with more than 1 elements to string\n"

I think we should leave this as was and file an issue to fix it. WDYT?

**Comment by lavalamp** [comment]
What exactly triggers that?

**Comment by lavalamp** [comment]
OK, since it's only a name change, LGTM. Can you file an issue?

**Comment by wojtek-t** [comment]
The issue is here:
https://github.com/kubernetes/kubernetes/issues/23849

I've re-pushed the old version (without returning error on length of slice >1). Readding lgtm.

**Comment by wojtek-t** [comment]
This is trivial renaming PR - merging manually to unblock more changes to generator.

**Pull request merged**